src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupServiceImpl.java

```diff
@@ -90,34 +90,37 @@ public LinkedHearingGroupServiceImpl(HearingRepository hearingRepository,
 
 
     @Override
+    @Transactional
     public HearingLinkGroupResponse linkHearing(HearingLinkGroupRequest hearingLinkGroupRequest) {
+        //POST
         linkedHearingValidator.validateHearingLinkGroupRequest(hearingLinkGroupRequest, null);
         LinkedGroupDetails linkedGroupDetails =
             linkedHearingValidator.updateHearingWithLinkGroup(hearingLinkGroupRequest);
-
         LinkedHearingGroup linkedHearingGroup = processRequestForListAssist(linkedGroupDetails);
 
         try {
             futureHearingRepository.createLinkedHearingGroup(objectMapperService
                                                                  .convertObjectToJsonNode(linkedHearingGroup));
             log.debug(LIST_ASSIST_SUCCESSFUL_RESPONSE);
-            linkedGroupDetailsRepository
-                .updateLinkedGroupDetailsStatus(linkedHearingGroup.getLinkedHearingGroup().getGroupClientReference(),
-                                                "ACTIVE");
+            linkedGroupDetails.setStatus("ACTIVE");
+            linkedGroupDetailsRepository.save(linkedGroupDetails);
         } catch (BadFutureHearingRequestException requestException) {
-            process400ResponseFromListAssistForLinkedHearing(
-                linkedHearingGroup.getLinkedHearingGroup().getGroupClientReference(), hearingLinkGroupRequest);
+            deleteLinkedHearingGroups(
```

Comment:
Will this not get rolled back on the exception? Do we need to delete? We are in the transaction here. As well cannot see verification in the test for the delete covered.